### ISSUE TITLE
Multiplayer Behaviour

### ISSUE THREAD

**Issue by ManuelSegura** [abearcodes/Valheim#2]
Probably know but anyway, this is what happens.

- I have the mod installed, my friend don't.
- There is a chest with 20 wood.
- I craft wooden arrows picking materials remotely through the use of this mod.
- When I check the chest, I see 12 wood.
- When friend checks the chest, he sees 20 wood.
- If either of us move the stack around, the final value we both see will be the value that the player who moved the stack was seeing.

Haven't check the code myself but there should be a way to sync the stacks. Maybe even just picking and dropping used stacks in the same spot after making crafting something.

**Comment by attckdog** [comment]
@ManuelSegura  You're going to have really weird behavior if all players aren't using the same codebase. I'd start by getting them on the same mod list.

**Comment by abearcodes** [comment]
Yup. Already replicated it and found a workaround. It's not ideal as all the code in the game tries to assume the inventories are basically "locked". 

The workaround is to force send the new state of the containers with the crafter as the owner. If the other player has the inventory open it will briefly flash, but that's the trade off.

**Comment by abearcodes** [comment]
Implemented partial fix. Should sync situations as described by @ManuelSegura  where people looking at the same chest see different items. Still need to fix when chest is actually in use. Keeping issue open.

**Comment by ManuelSegura** [comment]
Don't really know what "actually in use" means but can confirm that my friend, looking at chest while I craft, experience a flash (close > open) and see the new values correctly.

**Comment by abearcodes** [comment]
Didn't have the time to release a proper fix still, but some behaviors have already changed. 

There's a simple anti-duping mechanism in place now. As of 1.0.7 (just released), if you are not the last "owner" of a boat or ship (based on game logic), it will not consider that container. Therefore, pulling a cart and crafting should work ok. Steering a ship and crafting... eh... depends. If you are the first person that stepped on the ship, you are the owner. If somebody else opened the container on the ship, then they are the owner.

I do have a fix queued that will make all of these issues go away, but I don't have the time to work on it right now. At least these fixes should make the experience better! 

Keeping the issue open until a proper fix (or at least what I consider one to be) is implemented.

**Comment by ManuelSegura** [comment]
Posting here because probably is part of the same issue. Just observed this behaviour:

> Friend is looking at chest that contains one stack of honey
> I place a chest using this mod, so wood is picked from some random chest
> Friend experience flashing inventory, then immediately exists the chest menu
> The new chest is placed correctly
> The honey chest is now in "this chest is opened by someone else" state permanently for me
> My friend can unlock the chest by entering and exiting the honey chest

**Comment by abearcodes** [comment]
@ManuelSegura  thank you for reporting. Yeah, I can understand that it's inconvenient, but at least it prevents issues where people were desynced in terms of items. I'll prioritize this issue when I have more time for modding

**Comment by nanonull** [comment]
Is there a way to detect "locked" chests described in comment above ("this chest is opened by someone else" state ) and unlock them in coroutine with timeout maybe?

somewhere {
  // detect chest lock
  UnlockChest();
}
IEnumerator UnlockChest()
    {
        // suspend execution for 5 seconds
        yield return new WaitForSeconds(5);
        // set chest to "unlocked" state
    }

Must have mod

**Comment by ManuelSegura** [comment]
Hello again. After a while without experiencing this, today I had it happen. The same problem as the first post I mean. Me and my friends are seeing different things in the same chest and can easily dupe materials without knowing or wanting to. No other mods that affect chests are installed so I guess it's not because of incompatibilities.
